### DISCUSSION TITLE
How to implement UE's interfaces?

### DISCUSSION THREAD

**Discussion by Cthutq295** [hiperbou/kotlin-unreal#36] [Q&A]
@hiperbou Hello, I have a question about how to impement unreal engine interfaces like this:
https://docs.unrealengine.com/4.26/en-US/API/Runtime/AIModule/Perception/IAISightTargetInterface/CanBeSeenFrom/
Writing following code:
class AdventureCharacter: KotlinObject(), AISightTargetInterface {
override fun CanBeSeenFrom( ... ) { ... }
}
Would result in error "Only one class may appear in a supertype list" and "This type has a constructor, and thus must be initialized here" etc. Any ideas how to do it correctly ? I couldn't find anything simillar to my case in your example project.

**Comment by hiperbou** [comment]
You can only use functions that are visible from Blueprints. 
AISightTargetInterface::CanBeSeenFrom is only available from C++, so it's not visible neither in a Blueprint or Kotlin code.

On the other hand, the error you got is because Kotlin doesn't allow to do multiple inheritance, probably the definitions of AISightTargetInterface and others should be modified to specify that those are interfaces, so it will allow you to use it, although in this particular case, that interface has no functions, as they are only useable from C++.
